topics/integrate-in-existing-app.md

```diff
@@ -1,242 +1,423 @@
-is very useful to create such an architecture. 
+In your sample Android application, the business logic is stored in the package `com.jetbrains.simplelogin.androidapp.data`.
+Your future iOS application will use the same logic, so make it cross platform.
 
-A module should:
-*   Be a simple interface describing its inputs and outputs.
-*   Be in charge of a simple describable responsibility, such as database access, file management, a network API, or 
-credentials management.
+![Business logic to share](business-logic-to-share.png){width=350}
 
-To check if a module is ready for sharing, answer these two questions:
-*   Can it be tested with mock dependencies?
-*   Can it be mocked?
+### Create a shared module for cross-platform code
 
-If you can answer _yes_ to both, then you can share the module.
+The cross-platform code that is used for both iOS and Android _lives_ in the shared module.
+KMM provides a special wizard for creating such modules.
 
-### Identify modules to share
+In your Android project, create a KMM shared module for your cross-platform code. Later you'll connect it to your existing Android and future iOS applications.
 
-You can share business modules that have some platform-specific code. However, don't share a module that 
-mainly contains platform-specific code, because it would be easier to maintain two specific versions of the module, one for Android and one for iOS.
+1. In Android Studio, click **File** | **New** | **New Module**.
 
-For example, a module that reads or writes from the device storage contains a lot of platform code like APIs to access files. 
-So it is better if each platform project maintains its own version of the module by implementing the same interface.
+2. In the list of templates, select **KMM Shared Module**, enter the module name `shared` and select the **Generate packForXcode Gradle task** checkbox.  
+    This task is required for connecting the shared module to the iOS application.
 
-On the other hand, a module that manages credentials may contain some platform-specific code (such as for encryption), 
-but it mostly provides the logic that is common for both platforms. That's why it's a perfect candidate for sharing.
+   ![KMM shared module](kmm-module-wizard-1.png)
 
-You can also choose to share the UI behavior using the [Model-View-Presenter (_MVP_)](#mvp-for-legacy-ui-frameworks) 
-or the [Model-View-Intent (_MVI_)](#mvi-for-declarative-ui-frameworks) pattern. These patterns:
+3. Click **Finish**.
 
-*   Make a clear distinction between the UI and presentation layers.
-*   Are completely decoupled from the UI platform.
-*   Are easily testable without a UI environment.
+The wizard creates the KMM shared module, updates configuration files, and creates files with classes that demonstrate multiplatform benefits.
+You can learn more about the [KMM project structure](discover-kmm-project.md).
+    
+### Depend on the shared module in your Android application
```

Comment:
The heading sounds weird to me.

It seems like you can unite this section with the previous one — **Create a shared module ...**. 
The meaning will remain the same.